miqi/agent/skills.py

```
@@ -24,10 +74,28 @@ def __init__(self, workspace: Path, builtin_skills_dir: Path | None = None):
         self.builtin_skills = builtin_skills_dir or BUILTIN_SKILLS_DIR
         # #729: metadata/content 缓存——一次摘要构建内同一技能被查 450 次，
         # 每次无缓存都重读文件 + 全树 glob，实测单回合 5.7s。
-        self._meta_cache: dict[str, dict | None] = {}
+        # #859: meta_cache 提升为进程级共享（所有实例共享 frontmatter 解析结果）；
+        # 正文 content_cache 仍保持实例级（渐进式披露，正文按需读取）。
+        self._index = get_skill_index(workspace, self.builtin_skills)
+        self._meta_cache: dict[str, dict | None] = self._index.meta_cache
         self._content_cache: dict[str, str | None] = {}
         # nested 技能 name→SKILL.md 索引（懒构建，替代 load_skill 里的全树 glob）
         self._nested_index: dict[str, Path] | None = None
+        # 进程级索引的代际快照——检测磁盘变更，变了就清本实例的正文缓存
```

Ruff (0.16.2)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)